OpenRA.Mods.Common/Widgets/Logic/Ingame/ReplayControlBarLogic.cs

```diff
@@ -51,49 +56,88 @@ public ReplayControlBarLogic(Widget widget, World world, OrderManager orderManag
 
 				var playButton = widget.Get<ButtonWidget>("BUTTON_PLAY");
 				playButton.IsVisible = () => world.Timestep == 0 || orderManager.NetFrameNumber >= replayNetTicks;
-				playButton.OnClick = () => world.Timestep = (int)Math.Ceiling(originalTimestep * multipliers[speed]);
+				playButton.OnClick = () => world.Timestep = (int)Math.Ceiling(originalTimestep * multipliers[currentSpeed]);
 				playButton.IsDisabled = () => orderManager.NetFrameNumber >= replayNetTicks;
 
 				var slowButton = widget.Get<ButtonWidget>("BUTTON_SLOW");
-				slowButton.IsHighlighted = () => speed == PlaybackSpeed.Slow;
+				slowButton.IsHighlighted = () => currentSpeed == PlaybackSpeed.Slow;
```

Comment:
You can simplify things by binding `slowButton.GetKey = () => Game.Settings.Keys.ReplaySpeedSlowKey` (and similar for the other buttons).  Most of the other changes in this file can then go away.